CONSTRAINTS.md

```diff
@@ -0,0 +1,49 @@
+# Goals & Constraints
+
+Common Lifecycle Enumeration (CLE) aims to standardize the communication of software and hardware component lifecycle events in a structured, machine-readable format.
+
+Let’s explore some of the constraints and goals below.
+
+## 1. Balance Structure and Flexibility
+
+One of main tensions with designing the CLE is providing enough structure to make lifecycle data consistent and machine-readable, while allowing enough flexibility to cover a wide range of real-world scenarios.
+
+- MUST define a core set of common lifecycle events that the CLE will standardize. The standardized and shared understanding of key moments in a component's lifecycle is the necessary foundation for building interoperable lifecycle management and monitoring tooling.
+- MUST standardize how these events are represented in a formal schema. The focus at this stage is identifying the set of events to include in the first version of the specification and defining them. The selection of lifecycle events should be driven by
+  - Prevalence - How universally they apply across different component types and ecosystems.
+  - Impact - How much they influence risk and management of a component.
+- The events SHOULD enable component consumers to make informed decisions about risk, supportability, and upgrades.
+- The events MUST be unambiguous and factual in nature, not speculative or subjective.
+
+As the CLE is put into practice, it's likely that additional events will be identified that are important to standardize. There SHOULD be a clear process for proposing and ratifying new events into the core specification over time.
+
+## 2. Integration with existing systems and practices
+
+The CLE needs to work with the existing ecosystem of software supply chain standards, tooling, and processes. The spec MUST be designed for integration with existing tooling from day one.
+
+- MUST consider and define integration points with existing standards.
+    - PURL for component identification
+    - CycloneDX and SPDX for SBOMs
+    - TEA for data exchange and component aliasing
+- SHOULD be straightforward to implement incrementally within the tools and systems that organizations already use for building, dependency tracking, vulnerability management, SBOMs, etc. Adoption should not be an all-or-nothing affair.
+
+## 3. Handling identifier aliasing
```

Comment:
this is tricky, because it can be an attack vector ("popular package A is now malicious package B, i swear!")